### PR TITLE
[JENKINS-72343] Accept all 2xx and 3xx status codes to validate proxy in HTTP Proxy Configuration

### PR DESCRIPTION
**Pull request by lemeurherve**
The "Validate Proxy" button in "Manage > Plugins > Advanced settings > HTTP Proxy Configuration" currently accepts only `HTTP_OK` (200) as valid:
https://github.com/jenkinsci/jenkins/blob/51c5fa2367592f6fb6810f5eed89f88098ef0cf5/core/src/main/java/hudson/ProxyConfiguration.java#L579-L580

image

This small PR allows accepting all status code below `HTTP_BAD_REQUEST` (400) as valid responses and add the status code in the response to the user:

image

See [JENKINS-72343](https://issues.jenkins.io/browse/JENKINS-72343).

### Testing done

Checked that it works by running Jenkins locally with `mvn -pl war jetty:run`.
Ran a build with tests locally with `mvn -am -pl war,bom clean install`.
There is no test for this client side functionality, I don't think one should be added.

### Proposed changelog entries

- JENKINS-72343, Accept all 2xx and 3xx status codes to validate proxy in HTTP Proxy Configuration

(Not sure it worth being mentioned in changelog)

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by timja — APPROVED** [review]
seems reasonable, could maybe add the status code now given it's accepting more than 200?

**Comment by lemeurherve** [comment]
Added the status code in the response to the user:
image

**Review by NotMyFault — APPROVED** [review]
Thanks!

---

/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
